apis/nucleus/src/components/ActionsToolbar.jsx

```diff
@@ -90,24 +90,41 @@ const ActionsToolbar = ({
     show: false,
     anchorEl: null,
   },
+  focusHandler = null,
 }) => {
   const defaultSelectionActions = useDefaultSelectionActions(selections);
   const { itemSpacing } = useStyles();
-  const { translator } = useContext(InstanceContext);
+  const { translator, keyboardNavigation } = useContext(InstanceContext);
   const [showMoreItems, setShowMoreItems] = useState(false);
   const [moreEnabled, setMoreEnabled] = useState(more.enabled);
   const [moreActions, setMoreActions] = useState(more.actions);
   const [moreAlignTo, setMoreAlignTo] = useState(more.alignTo);
   const moreRef = useRef();
+  const actionsRef = useRef();
   const theme = useTheme();
   const dividerStyle = useMemo(() => ({ margin: theme.spacing(0.5, 0) }));
 
+  const getFirstEnabledButton = (reverse) => {
+    if (!actionsRef.current) return null;
+    const buttons = actionsRef.current.querySelectorAll('button:not(Mui-disabled)');
+    return reverse ? buttons[buttons.length - 1] : buttons[0];
+  };
+
   useEffect(() => () => setShowMoreItems(false), [popover.show]);
 
   useEffect(() => {
     setMoreEnabled(more.enabled);
   }, [more.enabled]);
 
+  useEffect(() => {
+    if (!focusHandler) return;
+
+    const focusFirst = () => getFirstEnabledButton(false)?.focus?.();
+    const focusLast = () => getFirstEnabledButton(true)?.focus?.();
+    focusHandler.on('focus_toolbar_first', focusFirst);
+    focusHandler.on('focus_toolbar_last', focusLast);
```

Comment:
Don't think I need to remove these, since they are attached to `focusHandler`, and not to the DOM somewhere

Reply:
In cell.js, yes possibly... I was getting errors when I added returned a function with `focusHandler.remove`

Reply:
I just checked, it does not. It is wrapped in a useEffect that only runs the first render, which is when the chart is mounted, not every time the toolbar shows up